app/models/mapping.rb

```diff
@@ -16,11 +16,45 @@ class Mapping < ApplicationRecord
 
   delegate :name, to: :user, prefix: true
 
+  after_create :after_created
+  after_update :after_updated
+  before_destroy :before_destroyed
+
```

Comment:
By moving the event publishing to these ActiveRecord hooks in the model file, we ensure that they will always get called, regardless of which controller or other code takes the `create` `update` or `destroy` actions